trees/redblacktree/redblacktree.go

```diff
@@ -192,6 +203,26 @@ func (tree *Tree) Right() *Node {
 	return parent
 }
 
+// Min returns the minimum key value pair of the AVL tree
+// or nils if the tree is empty.
+func (tree *Tree) Min() (interface{}, interface{}) {
```

Comment:
Left() returns this, do you think it would be more obvious to have a duplicate function Min()?

Also I learnt that returning the node rather than key-value pair when working with trees makes more sense, as opposed to maps where you expect the key-value pair.